install.sh

```diff
@@ -7,41 +7,63 @@ OPERATOR_CHANNEL=${OPERATOR_CHANNEL:-beta}
+  if [ -z "$SKIP_REGISTRY_LOGIN" ]; then
+    if [ -z "$OPERATOR_REGISTRY_USERNAME" ]; then
+      ${CONTAINER_RUNTIME} login $OPERATOR_REGISTRY_REF
+    else
+      ${CONTAINER_RUNTIME} login -u "$OPERATOR_REGISTRY_USERNAME" --pasword-stdin $OPERATOR_REGISTRY_REF <<<$OPERATOR_REGISTRY_PASSWORD
+    fi
   fi
-fi
 
-#3) Start a kubernetes cluster. Any Kubernetes cluster can be used as well. The script assumes that a Kubernetes is up and running and the user has logged into it.
+  #3) Start a kubernetes cluster. Any Kubernetes cluster can be used as well. The script assumes that a Kubernetes is up and running and the user has logged into it.
 
-#4) Enable [OLM](https://github.com/operator-framework/operator-lifecycle-manager) in the cluster by running the following command
-curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v${OLM_VERSION}/install.sh | bash -s v${OLM_VERSION}
-#On minikube you can alternatively install OLM by running `minikube addons enable olm`
+  #4) Enable [OLM](https://github.com/operator-framework/operator-lifecycle-manager) in the cluster by running the following command
+  curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v${OLM_VERSION}/install.sh | bash -s v${OLM_VERSION}
+  #On minikube you can alternatively install OLM by running `minikube addons enable olm`
 
-#5) Create a new image pull secret out of your local .docker/config.json file
-if [ -r "$DOCKER_CFG" ]; then
-  kubectl create secret generic sbo-operators-secrets -n olm --from-file=.dockerconfigjson=$DOCKER_CFG --type=kubernetes.io/dockerconfigjson
+  #5) Create a new image pull secret out of your local .docker/config.json file
+  if [ -r "$DOCKER_CFG" ]; then
+    kubectl create secret generic sbo-operators-secrets -n olm --from-file=.dockerconfigjson=$DOCKER_CFG --type=kubernetes.io/dockerconfigjson
 
-  #6) Add that pull secret to the default account in olm namespace
-  kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "sbo-operators-secrets"}]}' -n=olm
+    #6) Add that pull secret to the default account in olm namespace
+    kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "sbo-operators-secrets"}]}' -n=olm
+  fi
+}
+
+if [ "$(kubectl get crd clusterserviceversions.operators.coreos.com &>1 || echo false)" == "false" ]; then
+  echo no csv
+  INSTALL_OLM="true"
+  install_olm
 fi
 
+PACKAGESERVER_INFO=($(kubectl get clusterserviceversions.operators.coreos.com --all-namespaces| grep packageserver))
+echo PS_NAMESPACE=${PACKAGESERVER_INFO[0]}
+
+CATSRC_NAMESPACE="${CATSRC_NAMESPACE:-$(kubectl get catalogsources.operators.coreos.com --all-namespaces -o jsonpath='{.items[0].metadata.namespace}' --ignore-not-found=true)}"
+OPERATOR_NAMESPACE="${OPERATOR_NAMESPACE:-$(./hack/detect-operators-namespace.sh)}"
```

Comment:
it would be great that `install.sh` script does not depend on any other script in the repo, because potentially we would like to publish this script on our relase pages.